### DISCUSSION TITLE
Maybe it's an idea that every fragmentation strategy and every retention policy active in the server (so all the ones that are loaded) expose a shape and the validation of a view could then use these to verify if a valid view is being created.

### DISCUSSION THREAD

**Discussion by Tomvbe** [Informatievlaanderen/VSDS-LDESServer4J#754] [Ideas]
_Maybe it's an idea that every fragmentation strategy and every retention policy active in the server (so all the ones that are loaded) expose a shape and the validation of a view could then use these to verify if a valid view is being created._

_Originally posted by @WLefever-Cegeka in https://github.com/Informatievlaanderen/VSDS-LDESServer4J/issues/651#issuecomment-1571691415_

Every supported fragmentation is included in the current shacl shape. However if someone would write a custom fragmentation strategy, the shacl shape would not recoginize this fragmentation.

We could solve this with multiple solutions:

1. Add functionality to support adding shapes dynamically to the shacl
2. Remove the criteria that the fragmentationStrategy must be one of the supported strategies in the current shacl shape. I think removing this: 
`    	sh:or (tree:GeospatialFragmentation tree:PaginationFragmentation tree:SubstringFragmentation tree:TimebasedFragmentation) ;`
could work. @Yalz is the expert on this, so he may have a better idea.
3. Allow validation to be disabled

**Comment by sandervd** [comment]
This is a bit of a dependency issue, as now the domain depends on the fragmentation strategy implementations.
I'll open an issue, low prio, as the 'right' way would be that each fragmentation module provides the shape that validates its config.
